### PR TITLE
bpo-30246: fix several error messages which only mention bytes in struct

### PR DESCRIPTION
**Pull request by zhangyangyu**

### CODE REVIEW

File: Modules/_struct.c [hunk 1]
```diff
@@ -1734,8 +1735,8 @@ Struct_iter_unpack(PyStructObject *self, PyObject *buffer)
     }
     if (iter->buf.len % self->s_size != 0) {
         PyErr_Format(StructError,
-                     "iterative unpacking requires a bytes length "
-                     "multiple of %zd",
+                     "iterative unpacking requires a bytes-like object of "
+                     "length multiple of %zd",
```
Comment (lulouie) [inline Modules/_struct.c]: maybe `bytes-like object of its length is multiple of`?

Comment (louisom) [inline Modules/_struct.c]: maybe `bytes-like object of its length is multiple of`?

**Review by louisom — COMMENTED** [review]
small point, other LGTM

File: Modules/_struct.c [hunk 2]
```diff
@@ -1734,8 +1735,8 @@ Struct_iter_unpack(PyStructObject *self, PyObject *buffer)
     }
     if (iter->buf.len % self->s_size != 0) {
         PyErr_Format(StructError,
-                     "iterative unpacking requires a bytes length "
-                     "multiple of %zd",
+                     "iterative unpacking requires a buffer of "
```
Comment (serhiy-storchaka) [inline Modules/_struct.c:1738]: "length" is lost.

**Review by serhiy-storchaka — COMMENTED** [review]
Don't haste with merging this patch. I'll try to find the past discussion about the similar issue.

File: Modules/_struct.c (hunk 2 above)
Comment (zhangyangyu) [reply Modules/_struct.c:1738]: I delibrately discard it. Now the error messages in struct all get a "requires a buffer of %d bytes" format.

Comment (vadmium) [reply Modules/_struct.c:1738]: “. . . a buffer of _a_ multiple of . . .” would read better to me. Or simpler: “requires a multiple of . . . bytes”.

Comment (zhangyangyu) [reply Modules/_struct.c:1738]: Done.

**Comment by zhangyangyu** [comment]
What's the status of this PR @serhiy-storchaka ?

**Review by AraHaan — APPROVED** [review]
I think this wording is alright.

**Comment by zhangyangyu** [comment]
Is it okay for me to merge this now @vadmium @serhiy-storchaka ?

**Comment by vadmium** [comment]
The changes look okay to me.

**Pull request merged**